urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/CycDataBaseAccess.java

```diff
@@ -0,0 +1,687 @@
+import edu.iastate.javacyco.Frame;
+import edu.iastate.javacyco.JavacycConnection;
+import edu.iastate.javacyco.Network;
+import edu.iastate.javacyco.Organism;
+import edu.iastate.javacyco.Network.Edge;
+import edu.iastate.javacyco.OrgStruct;
+import edu.iastate.javacyco.PtoolsErrorException;
+
+@SuppressWarnings({"rawtypes","unchecked"})
+public class CycDataBaseAccess implements PropertyChangeListener {
+	private ProgressMonitor progressMonitor;
+	private SwingWorker task;
+	
+	public static JavacycConnection conn;
+	private String host;
+	private int port;
+	private String userName;
+	private String password;
+	
+	public CycDataBaseAccess(String host, int port, String userName, String password) {
+		this.host = host;
+		this.port = port;
+		this.userName = userName;
+		this.password = password;
+		
+		init();
+	}
+	
+	private void init() {
+		conn = null;
+		if (connect()) {
+			try {
+				// Select the first organism as a default
+				ArrayList<OrgStruct> orgs = conn.allOrgs();
+				if (orgs != null && orgs.size() > 0) {
+					conn.selectOrganism(conn.allOrgs().get(0).getLocalID());
+				}
+			} catch (PtoolsErrorException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+	
+	// Initialize connection
+	private boolean connect() {
+		try {
+			if (userName == null || userName.equalsIgnoreCase("") || password == null) conn = new JavacycConnection(host, port); 
+			else conn = new JavacycConnection(host, port, userName, password);
+			return true;
+		} catch(Exception e) {
+			e.printStackTrace();
+		}
+		return false;
+	}
+	
+	public JavacycConnection getConnection() {
+		return conn;
+	}
+	
+	public boolean testConnection() {
+		if (conn == null) return false;
+		try {
+			ArrayList<OrgStruct> orgs = conn.allOrgs();
+			if (orgs != null && orgs.size() > 0) return true;
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return false;
+	}
+	
+	public void setTimeOut (int miliseconds) {
+		conn.setQueryTimeOutLength(miliseconds);
+	}
+	
+	// Change selected organism
+	public void selectOrganism(String organism) {
+		ArrayList<String> organisms = new ArrayList<String>();
+		for (OrgStruct org : getAvailableOrganisms()) organisms.add(org.getLocalID());
+		conn.selectOrganism(organism);
+	}
+	
+	public String getSelectedOrganism() {
+		return conn.getOrganismID();
+	}
+	
+	// Get all organisms available at the current connection
+	public ArrayList<OrgStruct> getAvailableOrganisms() {
+		try {
+			return conn.allOrgs();
+		} catch (PtoolsErrorException e) {
+			e.printStackTrace();
+		}
+		return new ArrayList<OrgStruct>();
+	}
+	
+	// Convert a frame into a printable string
+	public String frameToString(String frameID) {
+		try {
+			Frame frame = Frame.load(conn, frameID);
+			frame.update();
+			return frameToString(frame);
+		}
+		catch(PtoolsErrorException e) {
+			return null;
+		}
+	}
+	
+	// Convert a frame into a printable string.  Prints all values in local copy of the frame
+	public String frameToString(Frame frame) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.